### ISSUE TITLE
SM 5.0.12 can't access the keyboard on macOS Sierra

### ISSUE THREAD

**Issue by xsznix** [stepmania/stepmania#1549]
I'm trying to play StepMania after a laptop and OS upgrade and while it accepts input from a USB dance pad (PDP Omega 4X), it doesn't respond to any input from the laptop builtin keyboard or external USB keyboards.

Running SM in the terminal gives this output:

```
$ ./StepMania.app/Contents/MacOS/StepMania
StepMania5.0.12
Compiled 20160811 @ 00:05:39 (build 45e0787)
Log starting 2017-10-03 23:10:40
Loading window: macosx
Model: MacBookPro11,5 (8/8)
Clock speed 2.50 GHz
Mac OS X 10.12.6
Memory: 16.00 GB
/////////////////////////////////////////
WARNING: ReadFile(Save/ThemePrefs.ini): No such file or directory
/////////////////////////////////////////
Sound driver: AudioUnit
Lights driver: SystemMessage
Video renderers: 'opengl'
Paletted textures disabled: GL_EXT_paletted_texture missing.
OGL Vendor: ATI Technologies Inc.
OGL Renderer: AMD Radeon R9 M370X OpenGL Engine
OGL Version: 2.1 ATI-1.51.8
OGL Max texture size: 16384
OGL Texture units: 8
GLU Version: 1.3 MacOSX
OGL Extensions:
  GL_APPLE: aux_depth_stencil, client_storage, element_array, fence, float_pixels, flush_buffer_range,
    flush_render, object_purgeable, packed_pixels, pixel_buffer, rgb_422, row_bytes, specular_vector,
    texture_range, transform_hint, vertex_array_object, vertex_array_range, vertex_point_size,
    vertex_program_evaluators, ycbcr_422
  GL_ARB: color_buffer_float, depth_buffer_float, depth_clamp, depth_texture, draw_buffers,
    draw_elements_base_vertex, draw_instanced, fragment_program, fragment_program_shadow, fragment_shader,
    framebuffer_object, framebuffer_sRGB, half_float_pixel, half_float_vertex, imaging, instanced_arrays,
    multisample, multitexture, occlusion_query, pixel_buffer_object, point_parameters, point_sprite,
    provoking_vertex, seamless_cube_map, shader_objects, shader_texture_lod, shading_language_100, shadow,
    shadow_ambient, sync, texture_border_clamp, texture_compression, texture_compression_rgtc, texture_cube_map,
    texture_env_add, texture_env_combine, texture_env_crossbar, texture_env_dot3, texture_float,
    texture_mirrored_repeat, texture_non_power_of_two, texture_rectangle, texture_rg, transpose_matrix,
    vertex_array_bgra, vertex_blend, vertex_buffer_object, vertex_program, vertex_shader, window_pos
  GL_ATI: blend_equation_separate, blend_weighted_minmax, separate_stencil, texture_compression_3dc,
    texture_env_combine3, texture_float, texture_mirror_once
  GL_EXT: abgr, bgra, bindable_uniform, blend_color, blend_equation_separate, blend_func_separate, blend_minmax,
    blend_subtract, clip_volume_hint, debug_label, debug_marker, depth_bounds_test, draw_buffers2,
    draw_range_elements, fog_coord, framebuffer_blit, framebuffer_multisample, framebuffer_object,
    framebuffer_sRGB, geometry_shader4, gpu_program_parameters, gpu_shader4, multi_draw_arrays,
    packed_depth_stencil, packed_float, provoking_vertex, rescale_normal, secondary_color,
    separate_specular_color, shadow_funcs, stencil_two_side, stencil_wrap, texture_array,
    texture_compression_dxt1, texture_compression_s3tc, texture_env_add, texture_filter_anisotropic,
    texture_integer, texture_lod_bias, texture_mirror_clamp, texture_rectangle, texture_sRGB, texture_sRGB_decode,
    texture_shared_exponent, timer_query, transform_feedback, vertex_array_bgra
  GL_IBM_rasterpos_clip
  GL_NV: blend_square, conditional_render, depth_clamp, fog_distance, light_max_exponent, texgen_reflection,
    texture_barrier
  GL_SGIS: generate_mipmap, texture_edge_clamp, texture_lod
  GL_SGI_color_matrix
OpenGL Windowed 854x480 32 color 16 texture 60Hz Vsync SmoothLines
/////////////////////////////////////////
WARNING: Failed to open the interface. - (iokit/common) exclusive access and device already open(e00002c5,709)
/////////////////////////////////////////
/////////////////////////////////////////
WARNING: Failed to open the interface. - (iokit/common) exclusive access and device already open(e00002c5,709)
/////////////////////////////////////////
/////////////////////////////////////////
WARNING: Failed to open the interface. - (iokit/common) exclusive access and device already open(e00002c5,709)
/////////////////////////////////////////
/////////////////////////////////////////
WARNING: Failed to open the interface. - (iokit/common) exclusive access and device already open(e00002c5,709)
/////////////////////////////////////////
```

**Comment by dguzek** [comment]
This might be the same issue as #1323.  Can you check to see what the contents of your Keymaps.ini file are?  On macOS, it will be located at `~/Library/Preferences/StepMania 5/Keymaps.ini`.

**Comment by xsznix** [comment]
I'm using the same keymap I used on my old laptop:

```
[pump]
1_Back=
1_Center=
1_Coin=
1_DownLeft=
1_DownRight=
1_EffectDown=
1_EffectUp=
1_MenuDown=
1_MenuLeft=
1_MenuRight=
1_MenuUp=
1_Operator=
1_Select=
1_Start=
1_UpLeft=
1_UpRight=
2_Back=
2_Center=
2_Coin=
2_DownLeft=
2_DownRight=
2_EffectDown=
2_EffectUp=
2_MenuDown=
2_MenuLeft=
2_MenuRight=
2_MenuUp=
2_Operator=
2_Select=
2_Start=
2_UpLeft=
2_UpRight=

[dance]
1_Back=
1_Coin=
1_Down=Key_h:Joy1_B4
1_EffectDown=
1_EffectUp=
1_Left=Key_t:Joy1_B5
1_MenuDown=
1_MenuLeft=
1_MenuRight=
1_MenuUp=
1_Operator=
1_Right=Key_a:Joy1_B3
1_Select=:Joy1_B1
1_Start=:Joy1_B2
1_Up=Key_n:Joy1_B6
1_UpLeft=
1_UpRight=
2_Back=
2_Coin=
2_Down=Key_r
2_EffectDown=
2_EffectUp=
2_Left=Key_e
2_MenuDown=
2_MenuLeft=
2_MenuRight=
2_MenuUp=
2_Operator=
2_Right=Key_i
2_Select=
2_Start=Key_y
2_Up=Key_u
2_UpLeft=
2_UpRight=
```

(I'm not on QWERTY so the keyboard mappings may seem a bit strange.)

**Comment by xsznix** [comment]
(whoops, misclicked)

I don't think #1323 is related. Perhaps it might be related to the complaints about not being able to open the iokit devices?

**Comment by dguzek** [comment]
That's likely an indication of the problem, yeah.  I don't get any iokit warnings when I run SM5.1-new on my MBP.

**Comment by xsznix** [comment]
Ah yes, Karabiner-Elements is the culprit here. Thanks for the help!

**Comment by halo** [comment]
After upgrading from High Sierra to Mojave the problem disappeared for `5.0.12` and I could use the keyboard again. 

However, `5.1 beta2` still has this problem on Mojave. The keyboard is not detected at all.

**Comment by drewlander** [comment]
I have the same problem, I compiled it from git today and I have the same issue. The only thing that I know of that may affect/capture keys is hammerspoon, but even shutting it down it does not matter. What can I check to make sure I have everything correct? 
OSX: 10.14.3
I have tried stepmania 5, and 5.1 and as i mentioned before, a build from git
I have Keymaps.ini  in ' ~/Library/Application\ Support/StepMania\ 5' (and 5.1 and 5.2) and still no go. I am sure I am missing something but cannot figure out what.
